src/Chainweb/Mempool/InMem.hs

```diff
@@ -241,6 +255,12 @@ insertInMem cfg lock runCheck txs0 = do
     sizeOK tx = getSize tx <= maxSize
     maxRecent = _inmemMaxRecentItems cfg
     hasher = txHasher txcfg
+    ttlCheck tx =
+      case txMetadata txcfg tx of
+        TransactionMetadata (Time (TimeSpan creationTime)) (Time (TimeSpan expiryTime)) ->
+            creationTime >= 0 && expiryTime >= 0 && creationTime < expiryTime
```

Comment:
This is the wrong property -- should be

```
creationTime < now && expiryTime <= now
```

You'll have to get the current time at the top of the check

Reply:
done.

Reply:
:open_mouth: oops... I've made the necessary change.